packages/meta-harness/lib/proposer_backend.py

```
@@ -0,0 +1,500 @@
+    args: list[str],
+    *,
+    cwd: Path,
+    capture_output: bool,
+    text: bool,
+    timeout: int | float,
+    stdin: int,
+    env: dict[str, str],
+) -> subprocess.CompletedProcess:
+    stdout = subprocess.PIPE if capture_output else None
+    stderr = subprocess.PIPE if capture_output else None
+    process = subprocess.Popen(
```

Ruff (0.15.20)

[error] `subprocess` call: check for execution of untrusted input

(S603)